src/core/ext/filters/client_channel/resolver/dns/c_ares/dns_resolver_ares.cc

```diff
@@ -366,7 +368,9 @@ void AresDnsResolver::OnResolvedLocked(void* arg, grpc_error* error) {
   }
   r->resolved_result_ = result;
   ++r->resolved_version_;
-  r->MaybeFinishNextLocked();
+  if (!r->shutdown_initiated_) {
```

Comment:
I think this is the wrong place for this check -- it should be somewhere in the block above where we start the retry timer.  If we know that we're being shut down when this function runs, then we shouldn't schedule the retry timer in the first place.

Reply:
Done - changed where this check takes place so that we avoid scheduling a re-resolution timer if we're completing an unsuccessful query.